nautilus_trader/common/actor.pyx

```diff
@@ -3453,8 +3573,53 @@ cdef class Actor(Component):
                 continue
 
             quantity = float(position.signed_qty)
-            instrument_greeks = self.instrument_greeks_data(position_instrument_id)
+            instrument_greeks = self.instrument_greeks(position_instrument_id, flat_interest_rate,
+                                                       spot_shock, vol_shock, expiry_in_years_shock,
+                                                       use_cached_greeks, cache_greeks, publish_greeks,
+                                                       ts_event, position)
             position_greeks = quantity * instrument_greeks
+            # self.log.warning(f"{position_greeks=}")
             portfolio_greeks += position_greeks
 
         return portfolio_greeks
+
+    def subscribe_greeks(self, underlying: str = "", handler: Callable[[GreeksData], None] = None):
+        """
+        Subscribe to Greeks data for a given underlying instrument.
+
+        Parameters
+        ----------
+        underlying : str, default ""
+            The underlying instrument ID prefix to subscribe to.
+            If empty, subscribes to all Greeks data.
+        handler : Callable[[GreeksData], None], optional
+            The callback function to handle received Greeks data.
+            If None, defaults to the cache_greeks method.
+
+        Returns
+        -------
+        None
+
+        """
+        used_handler = handler or self.cache_greeks
+        self._msgbus.subscribe(
+            topic=f"data.GreeksData.instrument_id={underlying}*",
+            handler=used_handler
+        )
+
+    def cache_greeks(self, greeks_data: GreeksData):
```

Comment:
I think its a bit asymmetric to expose a helper just for greeks, and not the other data types?
Better if we added `add_greeks` to the cache at this point?